### PR TITLE
Even more "immutability" across classes

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Even more "immutability" across classes  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
Cool, green.

**Comment by dnkoutso** [comment]
Do not remove until #7505 lands

File: lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb [hunk 1]
```diff
@@ -66,72 +64,39 @@ def generate
...
 
-          @xcconfig
+          xcconfig
         end
 
         #---------------------------------------------------------------------#
 
         protected
 
-        # @return String the SWIFT_VERSION of the target being integrated
```
Comment (dnkoutso) [inline lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb]: nothing changed here just organized to make those `private` and then `protected`.

File: lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb [hunk 2]
```diff
@@ -66,72 +64,39 @@ def generate
             'SWIFT_INCLUDE_PATHS' => '$(inherited) ',
           }.merge(embedded_content_settings)
 
-          @xcconfig = Xcodeproj::Config.new(config)
```
Comment (dnkoutso) [inline lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb]: removes state from this generator, there is no need to hold on to the `xcconfig` to this instance after we have doen the work.

File: lib/cocoapods/generator/xcconfig/pod_xcconfig.rb [hunk 3]
```diff
@@ -60,27 +63,25 @@ def generate
             'SWIFT_INCLUDE_PATHS' => '$(inherited) ',
           }
 
-          @xcconfig = Xcodeproj::Config.new(config)
```
Comment (dnkoutso) [inline lib/cocoapods/generator/xcconfig/pod_xcconfig.rb]: again removed unnecessary holding of state here.

**Comment by dnkoutso** [comment]
#7505 landed and reviewed. Will merge on green.

**Pull request merged**